ft-fine-tuning-inference/inference.py

```diff
@@ -0,0 +1,103 @@
+import os
+import pathlib
+import sys
+scripts_dir = pathlib.Path(__file__).parent.resolve()
+sys.path.append(str(scripts_dir))
+import numpy as np
+import torch
+from transformers import Trainer
+from transformers import BertTokenizer, BertForSequenceClassification
+import re
+import pandas as pd
+
+
+def preprocess(text, stem=False):
+    # TEXT CLENAING
+    TEXT_CLEANING_RE = "@\S+|https?:\S+|http?:\S|[^A-Za-z0-9]+"
+
+    # Remove link,user and special characters
+    text = re.sub(TEXT_CLEANING_RE, ' ', str(text).lower()).strip()
+    tokens = []
+    for token in text.split():
+        if stem:
+            tokens.append(stemmer.stem(token))
+        else:
+            tokens.append(token)
+    return " ".join(tokens)
+
+# Create torch dataset
+class Dataset_test(torch.utils.data.Dataset):
+    def __init__(self, encodings, labels=None):
+        self.encodings = encodings
+        self.labels = labels
+
+    def __getitem__(self, idx):
+        item = {key: val[idx] for key, val in self.encodings.items()}
+        # item['labels'] = self.labels[idx]
+        print(item)
+        return item
+        # input_ids = torch.tensor(self.encodings['input_ids'])
+        # target_ids = torch.tensor(self.labels[idx])
+        # return {"input_ids": input_ids, "labels": target_ids}
+
+    def __len__(self):
+        return len(self.encodings["input_ids"])
+
+
+# Define predict function
+def predict(data):
```

Comment:
who call predict(data)?

Reply:
The webservice type of library yaml will call main function predict()

![image](https://user-images.githubusercontent.com/89594861/200460842-f0efed01-8d94-483d-a5d3-a46425c8e64f.png)